### PR TITLE
[Model monitoring] Add metrics endpoint to httpdb

### PR DESCRIPTION
**Pull request by tomerm-iguazio**
1) Add `/metrics` endpoint to httpdb.
2) Add wrapper in project to `list_model_endpoints`.

+ test for each feature.

[ML-8334](https://iguazio.atlassian.net/browse/ML-8334)

[ML-8334]: https://iguazio.atlassian.net/browse/ML-8334?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mlrun/db/httpdb.py [hunk 1]
```diff
@@ -3365,6 +3366,31 @@ def delete_project_secrets(
...
+        type: Literal["results", "metrics", "all"] = "all",
+    ) -> list[mm_endpoints.ModelEndpointMonitoringMetric]:
+        path = f"projects/{project}/model-endpoints/{endpoint_id}/metrics"
+        params = {"type": type}
+        error_message = (
+            f"Failed to get model endpoint monitoring metrics,"
+            f" endpoint_id: {endpoint_id}, project: {project}"
+        )
+        response = self.api_call(
+            "GET",
```
Comment (jond01) [inline mlrun/db/httpdb.py]: Use the enum

File: mlrun/db/httpdb.py [hunk 2]
```diff
@@ -3365,6 +3366,31 @@ def delete_project_secrets(
             params=params,
         )
 
+    def get_model_endpoint_monitoring_metrics(
+        self,
+        project: str,
+        endpoint_id: str,
+        type: Literal["results", "metrics", "all"] = "all",
+    ) -> list[mm_endpoints.ModelEndpointMonitoringMetric]:
```
Comment (jond01) [inline mlrun/db/httpdb.py:3375]: 1. What is returned when there are no metrics?
2. Missing docstring

File: tests/system/model_monitoring/test_app.py [hunk 3]
```diff
@@ -248,27 +248,25 @@ def _test_api_get_metrics(
         run_db: mlrun.db.httpdb.HTTPRunDB,
         type: typing.Literal["metrics", "results"] = "results",
     ) -> list[str]:
-        cls._logger.debug("Checking GET /metrics API", type=type)
-        response = run_db.api_call(
-            method=mlrun.common.types.HTTPMethod.GET,
-            path=f"projects/{cls.project_name}/model-endpoints/{ep_id}/metrics?type={type}",
+        cls._logger.debug(
+            "querying httpdb get_model_endpoint_monitoring_metrics", type=type
```
Comment (jond01) [inline tests/system/model_monitoring/test_app.py]:
```suggestion
            "Checking the metrics", type=type
```

File: mlrun/projects/project.py [hunk 4]
```diff
@@ -3390,6 +3390,28 @@ def set_model_monitoring_credentials(
...
 
+    def list_model_endpoints(
+        self,
+        model: Optional[str] = None,
+        function: Optional[str] = None,
+        labels: Optional[list[str]] = None,
+        start: str = "now-1h",
+        end: str = "now",
+        top_level: bool = False,
+        uids: Optional[list[str]] = None,
```
Comment (jond01) [inline mlrun/projects/project.py:3401]: Nice, though I'm not sure about the defaults of `start` and `end`.
Missing docstring

**Review by jond01 — CHANGES_REQUESTED** [review]

File: mlrun/db/httpdb.py (hunk 2 above)
Comment (tomerm-iguazio) [reply mlrun/db/httpdb.py:3375]: 1. Fixed to return empty list if there is no result.
2. Added, tnx!

File: mlrun/projects/project.py (hunk 4 above)
Comment (tomerm-iguazio) [reply mlrun/projects/project.py:3401]: This is also the default in httpdb.list_model_endpoints, so I did not change it here...

File: mlrun/db/httpdb.py [hunk 5]
```diff
@@ -3365,6 +3366,40 @@ def delete_project_secrets(
...
+        """
+        :param project: The name of the project.
```
Comment (jond01) [inline mlrun/db/httpdb.py]: A general line on this method can be nice in addition to the params

File: mlrun/db/httpdb.py [hunk 6]
```diff
@@ -3365,6 +3366,40 @@ def delete_project_secrets(
...
+            params=params,
+        )
+        monitoring_metrics = response.json()
+        if monitoring_metrics:
+            return [
+                mm_endpoints.ModelEndpointMonitoringMetric(**monitoring_metric)
+                for monitoring_metric in monitoring_metrics
+            ]
+        else:
+            return []
```
Comment (jond01) [inline mlrun/db/httpdb.py]: 1. Consider using `pydantic.parse_obj_as`:
   https://stackoverflow.com/a/61021183/27658872
   Just a recommendation, up to you.
2. The if-else is probably unnecessary as `monitoring_metrics = []` returns the expected result.

File: server/py/framework/rundb/sqldb.py [hunk 7]
```diff
@@ -1017,6 +1017,8 @@ def list_model_endpoints(
         start: str = "now-1h",
         end: str = "now",
         metrics: Optional[list[str]] = None,
+        top_level: bool = False,
+        uids: Optional[list[str]] = None,
     ):
```
Comment (jond01) [inline server/py/framework/rundb/sqldb.py]: Can be nice to add the return type

File: mlrun/projects/project.py [hunk 8]
```diff
@@ -3390,6 +3390,59 @@ def set_model_monitoring_credentials(
...
+            the specified key-value pairs or key existence.
+        :param start: The start time of the metrics. Can be represented by a string containing an RFC 3339 time, a
+                      Unix timestamp in milliseconds, a relative time (`'now'` or `'now-[0-9]+[mhd]'`, where
+                      `m` = minutes, `h` = hours, `'d'` = days, and `'s'` = seconds), or 0 for the earliest time.
+        :param end: The end time of the metrics. Can be represented by a string containing an RFC 3339 time, a
+                      Unix timestamp in milliseconds, a relative time (`'now'` or `'now-[0-9]+[mhd]'`, where
+                      `m` = minutes, `h` = hours, `'d'` = days, and `'s'` = seconds), or 0 for the earliest time.
+        :param top_level: if true will return only routers and endpoint that are NOT children of any router
+        :param uids: if passed will return a list `ModelEndpoint` object with uid in uids
+        """
```
Comment (jond01) [inline mlrun/projects/project.py]: Missing `:return:`

**Review by jond01 — APPROVED** [review]
Great job!

**Pull request merged**